File: week-03/ek_18258_queue.cpp
#include <iostream>
#include <queue>

using namespace std;

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    queue<int> q;
    string order;

    while(n--) {
        cin >> order;

        if(order == "push") {
            int temp;
            cin >> temp;
            q.push(temp);
        } else if (order == "pop") {
            if(q.empty())
                cout << -1 << "\n";
            else {
                cout << q.front() << "\n";
                q.pop();
            }
        } else if (order == "size") {
            cout << q.size() << "\n";
        } else if (order == "empty") {
            cout << (q.empty() ? 1 : 0) << "\n";
        } else if (order == "front") {
            cout << (q.empty() ? -1 : q.front()) << "\n";
        } else if (order == "back") {
            cout << (q.empty() ? -1 : q.back()) << "\n";
        }
   }

    return 0;
}

File: week-03/ek_18258_queue.py
import sys
from collections import deque
read = sys.stdin.readline

n = int(read().rstrip())
dq = deque()

for _ in range(n):
    ord = list(read().split())

    if ord[0] == "push":
        dq.append(int(ord[1]))
    elif ord[0] == "pop":
        print(dq.popleft() if dq else -1)
    elif ord[0] == "size":
        print(len(dq))
    elif ord[0] == "empty":
        print(0 if dq else 1)
    elif ord[0] == "front":
        print(dq[0] if dq else -1)
    elif ord[0] == "back":
        print(dq[-1] if dq else -1)
